File: DriveApp/app/src/main/java/com/example/driveapp/data/models/Driver.kt
package com.example.driveapp.data.models

import com.google.gson.annotations.SerializedName

data class Driver(
    @SerializedName("first_name") val firstName: String,
    @SerializedName("last_name") val lastName: String,
    @SerializedName("phone") val phone: String,
    @SerializedName("pin") val pin: String,
    @SerializedName("gender") val gender: String,
    @SerializedName("email") val email: String,
    @SerializedName("county") val county: String,
    @SerializedName("sub_county") val subCounty: String
)

data class LoginRequest(
    val phone: String,
    val pin: String
)



data class LoginResponse(
    val token: String,
    @SerializedName("driver") val driver: DriverWithId
)

data class DriverWithId(
    @SerializedName("id") val id: Int,
    @SerializedName("first_name") val firstName: String,
    @SerializedName("last_name") val lastName: String,
    @SerializedName("phone") val phone: String,
    @SerializedName("pin") val pin: String,
    @SerializedName("gender") val gender: String,
    @SerializedName("email") val email: String,
    @SerializedName("county") val county: String,
    @SerializedName("sub_county") val subCounty: String,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String,
    @SerializedName("status") val status: String
)

File: ride-hailing/app/Models/Driver.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Laravel\Sanctum\HasApiTokens;
use Illuminate\Notifications\Notifiable;

class Driver extends Authenticatable
{
    use HasApiTokens, HasFactory,Notifiable;

    protected $fillable = [
        'id',
        'first_name',
        'last_name',
        'phone',
        'pin', // Assuming it's hashed
        'gender',
        'email',
        'county',
        'sub_county',
        'status', // Holds 'online' or 'offline'
    ];

    // Relationship with Ride
    public function rides()
    {
        return $this->hasMany(Ride::class);
    }

    // Accessor for status
    public function getStatusAttribute($value)
    {
        return ucfirst($value); // Returns "Online" or "Offline"
    }

    // Helper method to check if driver is online
    public function isOnline()
    {
        return $this->status === 'online';
    }

//     // Helper method to set status as online
//     public function setOnline()
//     {
//         $this->update(['status' => 'online']);
//     }

//     // Helper method to set status as offline
//     public function setOffline()
//     {
//         $this->update(['status' => 'offline']);
//     }
}
